### PR TITLE
[dart][dart-dio] Fix collection and date default values not compiling

### PR DESCRIPTION
**Pull request by kuhnroyal**
* empty array dafault values are now correctly generated for dart-dio
* default arrays with values are generated empty for now, supporting arrays with values requires much more work
* date/dateTime defaults are now always null in order to prevent compile errors, this also requires much more work

Fixes https://github.com/OpenAPITools/openapi-generator/issues/8285

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

CC @swipesight (2018/09) @jaumard (2018/09) @josh-burton (2019/12) @amondnet (2019/12) @sbu-WBT (2020/12) @kuhnroyal (2020/12) @agilob (2020/12)

### CODE REVIEW

**Comment by agilob** [comment]
I think with this bug would be more sensible to make a major breaking change and untie dart-dio from built_value. It's a 3rd party library and the deeper this goes the more bugs and incompatibilities will cause.

**Comment by kuhnroyal** [comment]
I disagree, same things are broken/not supported in the normal Dart generator. This has nothing to do with built_value.

**Comment by kuhnroyal** [comment]
I mean, I am all for adding an option to switch the serialization/generation library if anyone wants to contribute but I don't see it happen anytime soon.

**Review by agilob — APPROVED** [review]

**Comment by kuhnroyal** [comment]
Build failures unrelated, some Docker pull failure.

**Comment by agilob** [comment]
Not specific to this generator, but in general, I think adding dependency on a 3rd party library to handle things as simple as lists and maps shouldn't have place in generated code. The worst thing about whole dart are hard dependencies on analyzer, it's a bad joke when I can't use latest dart-angular with dart.inject because there's a conflict on `dependencies` (not dev_dependencies!) between these packages and any attempt to fix/update analyzer causes 5 more packages to conflict after that. The only twitter client in Dart uses really old `http` package... not allowing me to update postgresql package! Google isn't the any good at updating their public repos making dependency-hell in Dart much, much worse than I've seen/heard in java, .net or ruby.

Approved because you're clearly responsible for this generator and know what's best for it!

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java [hunk 1]
```diff
@@ -126,6 +126,16 @@ public String getHelp() {
...
     public String toDefaultValue(Schema schema) {
         if (schema.getDefault() != null) {
+            if (ModelUtils.isArraySchema(schema)) {
+                return "ListBuilder()";
+            }
+            if (ModelUtils.isMapSchema(schema)) {
+                return "MapBuilder()";
+            }
+            if (ModelUtils.isDateSchema(schema) || ModelUtils.isDateTimeSchema(schema)) {
+                // this is currently not supported and would create compile errors
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java:136]: FYI. I'll add a warning in another PR.

**Pull request merged**